vissl/data/disk_dataset.py

```diff
@@ -87,13 +87,6 @@ def _load_data(self, path):
             # Avoid creating it over and over again.
             self.is_initialized = True
 
-        if self.cfg["DATA"][self.split]["DATA_LIMIT"] > 0:
```

Comment:
I am a little bit worried about moving this out to the ssl_dataset.py. The reason is because we have some internal data sources that don't apply the data limit in similar way and could break training for them. Since our intent is to support the data limit like this for the synthetic and disk_dataset, can we put the `init_subset()` in data helper and call the function from respective disk_dataset.py and `synthetic_dataset.py` ? any other thoughts/suggestions. Sorry if I am missing some piece :)

Reply:
Do you have some pointers to these internal data sources ?

I mostly moved the code to `ssl_dataset.py` for two reasons:
- I saw that everything was scheduled in `ssl_dataset` (but as you point out, I clearly missing something) and found it would be a good way to factorise the code and the feature (we cannot random sample twice)
- The balanced random sampling (same amount of sample by classes) requires to have access to the labels, and these labels are only available in ssl_dataset

Reply:
yes the suggestion was to remove it move it to synthetic_dataset and disk_dataset. Let me share some examples . 
1) Essentially https://www.internalfb.com/intern/diffusion/FBS/browsefile/master/fbcode/deeplearning/projects/ssl_framework/vissl/fb/data/fairstore_dataset.py?commit=beaa5fafc669889804877ed3efd62c9df87e1507&lines=103-104 doesn't have the concept of DATA_LIMIT 
2) an internal change in https://www.internalfb.com/intern/diffusion/FBS/browsefile/master/fbcode/deeplearning/projects/ssl_framework/vissl/fb/data/everstore_dataset.py?commit=beaa5fafc669889804877ed3efd62c9df87e1507&lines=162-165 will be needed (a diff from fbcode) to adapt to the new call :)